RuleTool.py

```diff
@@ -34,91 +34,68 @@ def extract_sentence_cloud(prefix: str) -> str:
         return prefix[sentence_start:]
 
 
-# Procedural Functions
-def read_file(file: str) -> str:
-    if not (file and Path(file).is_file()):
-        print("Invalid Path")
-        exit(1)
-    file_text = Path(file).read_text()
-    return file_text
+def enumerate_rules(file_text: str) -> dict:
+    rule_start = file_text.find("(0:")
+    rule_end = file_text.find("}", rule_start) + 1
+    rule = file_text[rule_start:rule_end]
+    rule_prefix = file_text[:rule_start]
+    rule_prefixes = rule_prefix.split("\n")
+    if rule_prefixes[-1].find("//") > -1:
+        return {
+            "text": None,
+            "rule_end": rule_end
+        }
+    weight = extract_weight(rule_prefixes[-1])
+    sentence_cloud = extract_sentence_cloud(rule_prefixes[-2])
+    rule = {
+        "text": rule,
+        "weight": weight,
+        "sentence_cloud": sentence_cloud,
+        "rule_end": rule_end
+    }
+    return rule
+
+
+def separate_rule_components(rule: dict) -> dict:
+    original_texts = rule["text"].split("=>")
+    rule["nodes"] = original_texts[0]
+    if len(original_texts) > 1:
+        rule["action"] = original_texts[1]
+    else:
+        rule["action"] = None
+    return rule
```

Comment:
(Related to earlier comment) Seeing this, I feel like you should create the rule object in it's own function or class, then populate it in these functions. That way it's much easier to know all possible keys without having to read all functions.